xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VideoLayerBridgeDRMPRIME.cpp

```diff
@@ -125,9 +128,11 @@ void CVideoLayerBridgeDRMPRIME::Unmap(CVideoBufferDRMPRIME* buffer)
       buffer->m_handles[i] = 0;
     }
   }
+
+  buffer->Unmap();
```

Comment:
Do you want to move the `DRM_IOCTL_GEM_CLOSE` into the `Unmap()` function in `CVideoBufferDRMPRIME`? It is not needed for VAAPI.

Reply:
I am not sure I want to move it to `Unmap()` unless keeping it here actually breaks VAAPI. All code references I can find points to that a `drmPrimeFDToHandle` handle should be `DRM_IOCTL_GEM_CLOSE` to release the reference hold.

I need to investigate more on how kernel handles reference count for different dma-buf imports, in case we can do some optimization or fix uneven reference count in decoder or kernel. Hopefully result of sush investigation can be sent in a follow up PR in near future.